src/Symfony/Component/Finder/Iterator/FilterIterator.php

```diff
@@ -18,9 +18,24 @@
  *
  * @author Alex Bogomazov
  */
-abstract class FilterIterator extends \FilterIterator
+abstract class FilterIterator extends \FilterIterator implements \RecursiveIterator
```

Comment:
But what if that's exactly what I want, excluding `/Tests$/` and having it exclude `.../Tests/foo.php`?
This give more power to the user, and we don't force what can be done does with the iterators.
More power means more way to get things wrong of course... I see no issue with that.